File: java/src/main/java/com/workshop/architecture/fitness/hexagon/inside/port/outbound/MembershipInvoiceDetails.java
package com.workshop.architecture.fitness.hexagon.inside.port.outbound;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

public record MembershipInvoiceDetails(UUID id, UUID customerId, UUID membershipId,
                                       UUID planId, LocalDate dueDate, String planTitle, BigDecimal planPrice) {
}

File: python/src/workshop_api/fitness/hexagon/inside/port/outbound/membership_invoice_details.py
from dataclasses import dataclass
from datetime import date
from decimal import Decimal
from uuid import UUID


@dataclass(frozen=True)
class MembershipInvoiceDetails:
    id: UUID
    customer_id: UUID
    membership_id: UUID
    plan_id: UUID
    due_date: date
    plan_title: str
    plan_price: Decimal
